sympy/solvers/ode.py

```diff
@@ -5770,116 +5770,130 @@ def ode_lie_group(eq, func, order, match):
+        for heuristic in heuristics:
+            try:
+                if not inf:
+                    inf = infinitesimals(eq, hint=heuristic, func=func, order=1, match=match)
+            except ValueError:
+                continue
+            else:
+                for infsim in inf:
+                    xiinf = (infsim[xi(x, func)]).subs(func, y)
+                    etainf = (infsim[eta(x, func)]).subs(func, y)
+                    # This condition creates recursion while using pdsolve.
+                    # Since the first step while solving a PDE of form
+                    # a*(f(x, y).diff(x)) + b*(f(x, y).diff(y)) + c = 0
+                    # is to solve the ODE dy/dx = b/a
+                    if simplify(etainf/xiinf) == h:
+                        continue
+                    rpde = f(x, y).diff(x)*xiinf + f(x, y).diff(y)*etainf
+                    r = pdsolve(rpde, func=f(x, y)).rhs
+                    s = pdsolve(rpde - 1, func=f(x, y)).rhs
+                    newcoord = [_lie_group_remove(coord) for coord in [r, s]]
+                    r = Dummy("r")
+                    s = Dummy("s")
+                    C1 = Symbol("C1")
+                    rcoord = newcoord[0]
+                    scoord = newcoord[-1]
+                    try:
+                        sol = solve([r - rcoord, s - scoord], x, y, dict=True)
+                        if sol == []:
+                            continue
+                    except NotImplementedError:
+                        continue
+                    else:
+                        sol = sol[0]
+                        xsub = sol[x]
+                        ysub = sol[y]
+                        num = simplify(scoord.diff(x) + scoord.diff(y)*h)
+                        denom = simplify(rcoord.diff(x) + rcoord.diff(y)*h)
+                        if num and denom:
+                            diffeq = simplify((num/denom).subs([(x, xsub), (y, ysub)]))
+                            sep = separatevars(diffeq, symbols=[r, s], dict=True)
+                            if sep:
+                                # Trying to separate, r and s coordinates
+                                deq = integrate((1/sep[s]), s) + C1 - integrate(sep['coeff']*sep[r], r)
+                                # Substituting and reverting back to original coordinates
+                                deq = deq.subs([(r, rcoord), (s, scoord)])
+                                try:
+                                    sdeq = solve(deq, y)
+                                except NotImplementedError:
+                                    tempsol.append(deq)
                                 else:
-                                    return [Eq(f(x), sol) for sol in sdeq]
+                                    if len(sdeq) == 1:
+                                        return Eq(f(x), sdeq.pop())
+                                    else:
+                                        return [Eq(f(x), sol) for sol in sdeq]
```

Comment:
Do we need to unpack it when len(sdeq)==1 or can we just return the list? I thought you said before that the list would get unpacked higher up in dsolve.

Reply:
I will care about this in next pr.